File: src/main/kotlin/com/itmo/microservices/demo/warehouse/impl/repository/WarehouseItemRepository.java
package com.itmo.microservices.demo.warehouse.impl.repository;

import com.itmo.microservices.demo.warehouse.impl.entity.WarehouseItem;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.UUID;

public interface WarehouseItemRepository extends JpaRepository<WarehouseItem, UUID> {
    @Override
    boolean existsById(UUID uuid);
    WarehouseItem findWarehouseItemById(UUID uuid);
}

File: src/main/kotlin/com/itmo/microservices/demo/warehouse/impl/repository/WarehouseItemRepository.kt
package com.itmo.microservices.demo.warehouse.impl.repository

import org.springframework.data.jpa.repository.JpaRepository
import com.itmo.microservices.demo.warehouse.impl.entity.WarehouseItem
import java.util.UUID

interface WarehouseItemRepository : JpaRepository<WarehouseItem?, UUID?> {
    override fun existsById(uuid: UUID): Boolean
    fun findWarehouseItemById(uuid: UUID?): WarehouseItem?
}